packages/ruby/lib/readme/metrics.rb

```diff
@@ -32,13 +35,18 @@ def call(env)
       har = Har.new(env, response, start_time, end_time, @filter)
       user_info = @get_user_info.call(env)
       payload = Payload.new(har, user_info, development: @development)
-
-      HTTParty.post(
-        ENDPOINT,
-        basic_auth: {username: @api_key, password: ""},
-        headers: {"Content-Type" => "application/json"},
-        body: payload.to_json
-      )
+      @@request_queue.push(payload.to_json)
+
+      if @@request_queue.ready_to_send?
```

Comment:
Is there a possible race condition here where two requests check if the queue is ready to send and then they both send out the same values? Maybe I'm getting ahead of myself 😆

Reply:
To mitigate this potential race condition, we pulled the queue-checking functionality into the `push` method. 

When the buffer is long enough, the first n number of payloads are pulled from the queue, according to the `buffer_length`. This ensures that we're only sending a particular set of payloads once.